### ISSUE TITLE
SC2312 and bash 5.2

### ISSUE THREAD

**Issue by pan64** [koalaman/shellcheck#2718]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2312)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
# shellcheck enable=check-extra-masked-returns
read -r a b v < <(echo this is something | grep nothing)
rc=$?       # return code of read
wait $!
rc2=$?     # return code of subprocess

```

#### Here's what shellcheck currently says:

SC2312 Consider invoking this command separately to avoid masking its return value (or use '|| true' to ignore).

#### Here's what I wanted or expected to see:

Actually the result of process substitution can be checked using wait and therefore the return value is not masked.
This solution does not work with the documented example: cd "$(command)/etc" you may say it is a different case.
Also it works only with bash 5.x (5.2 was tested), but does not work with bash 4.3.

**Comment by Silic0nS0ldier** [comment]
I had a go at implementing this, with little success (hardly surprising given it was my first time touching Haskell).

In lieu of actual handling I've documented `wait $!` as an alternate way to handle this under **Exceptions** in [shellcheck/wiki/SC2312](https://github.com/koalaman/shellcheck/wiki/SC2312/8aed6d66a7858ff45f8dd6a34237d3ca56b1fd32)
